packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/handlers/GetOmniboxCommandsEventHandler.java

```diff
@@ -0,0 +1,113 @@
+import org.eclipse.sirius.components.collaborative.messages.ICollaborativeMessageService;
+import org.eclipse.sirius.components.core.api.ErrorPayload;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IInput;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.representations.Message;
+import org.eclipse.sirius.components.representations.MessageLevel;
+import org.springframework.stereotype.Service;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import reactor.core.publisher.Sinks;
+
+/**
+ * The event handler to retrieve omnibox commands in the context of an editing context.
+ *
+ * @author gcoutable
+ */
+@Service
+public class GetOmniboxCommandsEventHandler implements IEditingContextEventHandler {
+
+    private final IObjectService objectService;
+
+    private final ICollaborativeMessageService messageService;
+
+    private final IQueryService queryService;
+
+    private final Counter counter;
+
+    public GetOmniboxCommandsEventHandler(IObjectService objectService, ICollaborativeMessageService messageService, IQueryService queryService, MeterRegistry meterRegistry) {
+        this.objectService = Objects.requireNonNull(objectService);
+        this.messageService = Objects.requireNonNull(messageService);
+        this.queryService = Objects.requireNonNull(queryService);
+
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+    }
+
+    @Override
+    public boolean canHandle(IEditingContext editingContext, IInput input) {
+        return input instanceof GetOmniboxCommandsInput;
+    }
+
+    @Override
+    public void handle(Sinks.One<IPayload> payloadSink, Sinks.Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IInput input) {
+        this.counter.increment();
+
+        List<Message> messages = List.of(new Message(this.messageService.invalidInput(input.getClass().getSimpleName(), CreateChildInput.class.getSimpleName()),
+                MessageLevel.ERROR));
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, editingContext.getId(), input);
+        IPayload payload = null;
+
+        if (input instanceof GetOmniboxCommandsInput getOmniboxCommandsInput) {
+            List<OmniboxCommand> omniboxCommands = new ArrayList<>();
+
+            var queryBasedObjectsInput = new QueryBasedObjectsInput(UUID.randomUUID(), "aql:editingContext.allContents()->select(e | e.name.contains('" + getOmniboxCommandsInput.query() + "'))", Map.of("editingContext", editingContext));
+            var objectsPayload = this.queryService.execute(editingContext, queryBasedObjectsInput);
```

Comment:
This could do it for now but since some metamodels can provide search capabilities which would dramatically improve performances, we will have to perform this search ourselves very soon (in this iteration). Once the PR on REST services will be merged and this one too. We should use a common strategy to iterate on the whole resource set. In that PR, the implementation of `DefaultObjectRestService` does not match what I need either since it chooses to ignore resources for example.

We will probably have to upgrade `IContentService` and `IDefaultContentService` to provide `getAllContents` too.